src/main/java/net/glowstone/util/SoundUtil.java

```diff
@@ -0,0 +1,40 @@
+package net.glowstone.util;
+
+import net.glowstone.GlowWorld;
+import net.glowstone.entity.GlowPlayer;
+import org.bukkit.Location;
+import org.bukkit.Sound;
+
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class SoundUtil {
+
+    private static final ThreadLocalRandom rand = ThreadLocalRandom.current();
+
+    public static void playSoundAtLocationExcept(Location location, Sound sound, float volume, float pitch, GlowPlayer... exclude) {
+        if (location == null || sound == null) return;
+        GlowWorld world = (GlowWorld) location.getWorld();
+        double radiusSquared = Math.pow(volume * 16, 2);
+        world.getRawPlayers().stream().filter(player -> player.getLocation().distanceSquared(location) <= radiusSquared).filter(player -> !Arrays.asList(exclude).contains(player)).forEach(player -> player.playSound(location, sound, volume, pitch));
```

Comment:
world.getRawPlayers().stream() could use (`parallelStream()`)[http://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#parallelStream--] instead, since I believe it's safe to use it here (further elements in the list do not require the previous elements to be looped through).